src/rdkafka.h

```diff
@@ -2370,13 +2370,34 @@ size_t rd_kafka_queue_length (rd_kafka_queue_t *rkqu);
  *
  * librdkafka will maintain a copy of the \p payload.
  *
+ * @remark FD and callback IO event triggering are mutually exclusive.
```

Comment:
IO in this context refers to the file descriptor triggering.
I don't think the callback based triggering should be named IO since it doesn't really do any IO or can't be directly integrated with an IO scheduler in the way the fd-based one can.

So please revert your IO -> FD changes, and go with the following nomenclature:
IO event = FD
callback event = callback

Reply:
I see, makes sense. Thank you.